### PR TITLE
[CDAP-15919] Adds Dynamic plugin filters to http plugin

### PR DESCRIPTION
**Pull request by ajainarayanan**

### CODE REVIEW

File: widgets/HTTP-batchsource.json [hunk 1]
```diff
@@ -101,6 +102,60 @@
         }
       ]
     },
+    {
+      "label": "OAuth2",
+      "properties": [
+        {
+          "widget-type": "radio-group",
```
Comment (albertshau) [inline widgets/HTTP-batchsource.json]: existing, but for new plugins I think we should use the toggle widget for booleans.

Comment (elfenheart) [reply widgets/HTTP-batchsource.json]: done

File: widgets/HTTP-streamingsource.json [hunk 2]
```diff
@@ -502,5 +497,188 @@
...
+        {
+          "name": "refreshToken",
+          "type": "property"
+        }
+      ]
+    },
+    {
+      "name": "JSON/XML Formatting",
+      "condition": {
+        "expression": "format == 'json' || format == 'xml'"
```
Comment (bdmogal) [inline widgets/HTTP-streamingsource.json:656]: I was going to open a jira about this. Shouldn't we instead support operators such as 'in', 'not in', 'matches', 'not matches', 'contains', 'not contains'? For simple conditions like these, we shouldn't need to specify an expression right?

@elfenheart @albertshau

Comment (albertshau) [reply widgets/HTTP-streamingsource.json:656]: +1. I think the workaround today would be to have two filters, one for json and one for xml.

**Review by albertshau — APPROVED** [review]

File: widgets/HTTP-streamingsource.json (hunk 2 above)
Comment (bdmogal) [reply widgets/HTTP-streamingsource.json:656]: https://issues.cask.co/browse/CDAP-16023

Comment (ajainarayanan) [reply widgets/HTTP-streamingsource.json:656]: This is right now possible. https://github.com/TomFrost/Jexl#a-note-about-in

**Pull request merged**